packages/syft/tests/syft/core/tensor/passthrough_test.py

```diff
@@ -1,12 +1,970 @@
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__rshift__(tensor_b)
+    result_b = tensor_a.__rshift__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__rrshift__() -> None:
+    data_a = np.array([1, 2, -1], dtype=np.int32)
+    data_b = torch.tensor([0, 2, 1], dtype=torch.int32)
+    expected = torch.tensor([0, 8, -1], dtype=torch.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__rrshift__(tensor_b)
+    result_b = tensor_a.__rrshift__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__pow__() -> None:
+    data_a = np.array([1, 2, -1], dtype=np.int32)
+    data_b = np.array([0, 2, 1], dtype=np.int32)
+    expected = np.array([1, 4, -1], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__pow__(tensor_b)
+    result_b = tensor_a.__pow__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__rpow__() -> None:
+    data_a = np.array([1, 2, 1], dtype=np.int32)
+    data_b = torch.tensor([0, 2, -1], dtype=torch.int32)
+    expected = torch.tensor([0, 4, -1], dtype=torch.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__rpow__(tensor_b)
+    result_b = tensor_a.__rpow__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__divmod__() -> None:
+    data_a = np.array([1, 2, 3], dtype=np.int32)
+    data_b = np.array([1, 5, 4], dtype=np.int32)
+    expected = np.array([0, 1, 3], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=data_b)
+    tensor_c = PassthroughTensor(child=expected)
+    result_a = tensor_a.__divmod__(tensor_b)
+    result_b = tensor_a.__divmod__(data_b)
+
+    assert result_a == tensor_c
+    assert result_b == tensor_c
+
+
+def test__neg__() -> None:
+    data_a = np.array([1, 0, -1], dtype=np.int32)
+    expected = np.array([-1, 0, 1], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=expected)
+
+    assert tensor_a.__neg__() == tensor_b
+
+
+def test__invert__() -> None:
+    data_a = np.array([-1, 0, 1], dtype=np.int32)
+    expected = np.array([0, -1, -2], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+    tensor_b = PassthroughTensor(child=tensor_a.__invert__())
+    tensor_c = PassthroughTensor(child=expected)
+
+    assert tensor_b == tensor_c
+
+
+def test__index__() -> None:
+    data_a = np.array([1, 2, 3], dtype=np.int32)
+    tensor_a = PassthroughTensor(child=data_a)
+
+    assert tensor_a[2].__index__() == 3
```

Comment:
I think this needs to be a np.array scalar:
```
>>> data_a = np.array([1, 2, 3], dtype=np.int32)
>>> data_a
array([1, 2, 3], dtype=int32)
>>> data_a.__index__()
TypeError: only integer scalar arrays can be converted to a scalar index

>>> a = np.array(1)
>>> a
array(1)
>>> a.__index__()
1
```

They are weird unit scalar values wrapped as an np.array so that operations between them and other np.arrays can chain properly and output more np.arrays I believe.

Reply:
Changed test to: 

def test__index__() -> None:
    data_a = np.array(1)
    tensor_a = PassthroughTensor(child=data_a)

    assert tensor_a.__index__() == 1